packages/backend/src/tokens/request.ts

```diff
@@ -174,6 +175,13 @@ ${error.getFullMessage()}`,
       // Right now the only usage of passing in different headers is for multi-domain sync, which redirects somewhere else.
       // In the future if we want to decorate the handshake redirect with additional headers per call we need to tweak this logic.
       const handshakeHeaders = headers ?? buildRedirectToHandshake();
+
+      // Chrome aggressively caches inactive tabs. If we don't set the header here,
+      // all 307 redirects will be cached and the handshake will end up in an infinite loop.
+      if (handshakeHeaders.get(constants.Headers.Location)) {
+        handshakeHeaders.append(constants.Headers.CacheControl, 'no-store');
```

Comment:
🔧 I believe it's safer to overwrite any existing values here (i.e. `set`), just in case some code earlier in the path (e.g. another middleware) already sets `Cache-Control: public`.

Reply:
Good point